### PR TITLE
feature/add ngrx

### PR DESCRIPTION
**Pull request by matigrieben**

### CODE REVIEW

File: src/DonateTo.WebApi/Startup.cs [hunk 1]
```diff
@@ -75,7 +75,9 @@ private void SetupCorsPolicyAction(CorsOptions options)
                     DonateToCorsPolicy,
                     builder =>
                     {
-                        builder.WithOrigins(domainsAllowed.Split(';'));
+                        builder.WithOrigins(domainsAllowed.Split(';'))
+                            .AllowAnyHeader()
+                            .AllowAnyMethod();
```
Comment (matigrieben) [inline src/DonateTo.WebApi/Startup.cs:80]: This is because the browser send OPTIONS, before sending the POST request, because of CORS.

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/http-error.interceptor.ts [hunk 2]
```diff
@@ -2,14 +2,12 @@ import {Injectable} from '@angular/core';
...
 import {throwError} from 'rxjs';
 import {catchError} from 'rxjs/internal/operators';
-import {ErrorService} from '../error.service';
 
 @Injectable({
   providedIn: 'root'
 })
 
 export class HttpErrorInterceptor implements HttpInterceptor {
-    constructor(private errorService: ErrorService) {}
```
Comment (matigrieben) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/http-error.interceptor.ts:12]: Not used

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/base-http-client.service.ts [hunk 3]
```diff
@@ -63,10 +57,7 @@ export class BaseHttpClientService<T extends BaseModel> {
    * @returns Observable of T
    */
   update(item: T): Observable<T> {
-    return this.httpClient.put<T>(`${this.url}/${this.endpoint}/${item.id}`, JSON.stringify(item), this.httpOptions)
-    .pipe(
-      map(data => this.serializer.fromJson(data) as T)
-      );
```
Comment (matigrieben) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/base-http-client.service.ts:69]: Serializer not used, since we return a simple object as Observer

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/base-http-client.service.ts [hunk 4]
```diff
@@ -50,10 +47,7 @@ export class BaseHttpClientService<T extends BaseModel> {
    */
   create(item: T): Observable<T> {
     return this.httpClient
-    .post<T>(`${this.url}/${this.endpoint}`, JSON.stringify(item), this.httpOptions)
-    .pipe(
-      map(data => this.serializer.fromJson(data) as T)
-      );
```
Comment (matigrieben) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/base-http-client.service.ts:56]: Serializer not used, since we return a simple object as Observer

File: src/DonateTo.WebApi/V1/Controllers/BaseApiController.cs [hunk 5]
```diff
@@ -8,11 +9,11 @@ namespace DonateTo.WebApi.V1.Controllers
     [ApiController]
     public abstract class BaseApiController<T> : ControllerBase where T : class
     {
-        private IBaseService<T> _baseService  { get; set; }
+        private IBaseService<T> BaseService  { get; set; }
```
Comment (nicolasgranata) [inline src/DonateTo.WebApi/V1/Controllers/BaseApiController.cs:12]: revert this change, it should be _baseService
